### PR TITLE
Add Github Worfklow coverage to update values.yaml and Chart.yaml dependency versions

### PR DESCRIPTION
**Pull request by jvoravong**
Description
- Needed to fix an issue with update_instrumentation_dependencies.yaml not getting the latest version numbers properly.
- Added .github/workflows/update_chart_dependencies.yaml so we could automate updating subchart dependencies.
- Dependabot makes very similar updates compared to these, but Dependabot does not support updating dependencies like these yet.

Testing
- Was able to successfully test and validate both workflows in my forked repo.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -14,6 +14,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 - Update PodDisruptionBudgets API version to allow both `policy/v1beta1` and `policy/v1` [#835](https://github.com/signalfx/splunk-otel-collector-chart/pull/835)
 - Update clusterrole to allow collector to check for the `aws-auth` configmap in EKS clusters [#840](https://github.com/signalfx/splunk-otel-collector-chart/pull/840)
 - Add support to create default Instrumentation for operator based auto-instrumentation [#836](https://github.com/signalfx/splunk-otel-collector-chart/pull/836)
+- Add Github workflow coverage to update values.yaml and Chart.yaml dependency versions [#852](https://github.com/signalfx/splunk-otel-collector-chart/pull/852)
```
Comment (dmitryax) [inline CHANGELOG.md:17]: I don't think we need to put tooling updates in the change log for end users

Comment (jvoravong) [reply CHANGELOG.md]: Updated

File: .github/workflows/update_chart_dependencies.yaml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+          if [ "$LATEST_VER" == "$DEP_PATH" ]; then
+            echo We are already up to date. Nothing else to do.
+          else
+            echo 'Looks like we need to update...'
+            echo Updating to new version in chart.yaml
+            echo "NEED_UPDATE=1" >> $GITHUB_OUTPUT
+
+            # Update the version in chart.yaml using yq and jq
+            DEP_LINE=$(yq eval ".dependencies | keys | map(tonumber) | map(select(. != null)) | map(select(. < 10000)) | map(. + 1)" $CHART_YAML | jq ".[] | select(.[\"name\"] == \"${{ matrix.repo }}\")")
+            sed -i "${DEP_LINE}s/$DEP_PATH/$LATEST_VER/" $CHART_YAML
```
Comment (jinja2) [inline .github/workflows/update_chart_dependencies.yaml:45]: Could be updated with yq directly with cmd like this
```
yq e -i '.dependencies[] |= (select(.name == \"${{ matrix.repo }}\").version = env(LATEST_VER))' $CHART_YAML
```

File: .github/workflows/update_instrumentation_dependencies.yaml [hunk 3]
```diff
@@ -14,6 +15,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
+        # Currently this workflow will update the listed operator instrumentatipon dependencies in values.yaml
```
Comment (jinja2) [inline .github/workflows/update_instrumentation_dependencies.yaml:18]:
```suggestion
        # Currently this workflow will update the listed operator instrumentation dependencies in values.yaml
```

File: .github/workflows/update_chart_dependencies.yaml (hunk 2 above)
Comment (jvoravong) [reply .github/workflows/update_chart_dependencies.yaml:45]: Would like that yq solution more. Ran into yq update issues in this environment, I'm going to leave this with sed for now.

**Review by jinja2 — APPROVED** [review]

**Review by dmitryax — APPROVED** [review]

**Pull request merged**